stack_test.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaller(t *testing.T) {
+	s := NewStackTrace(0)
+	assert.NotEmpty(t, s.StackTrace)
+
+	assert.Equal(t, fmt.Sprintf("%s", s), "[stack_test.go testing.go asm_amd64.s]")
```

Comment:
Let's try to find a better and easy solution to what we already have.

If we cannot find anything better or if all solutions are too complicated, then fine, let's keep it as it is right now.